src/client/cockpit-beiboot

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import argparse
+import asyncio
+import base64
+import importlib.resources
+import json
+import logging
+import lzma
+import os
+import shlex
+import sys
+from pathlib import Path
+from typing import Iterable, Optional, Sequence
+
+from cockpit import polyfills
+from cockpit._vendor import ferny
+from cockpit._vendor.bei import beipack, bootloader
+from cockpit.bridge import setup_logging
+from cockpit.channel import ChannelRoutingRule
+from cockpit.channels import PackagesChannel
+from cockpit.data import read_cockpit_data_file
+from cockpit.packages import Package, Packages
+from cockpit.peer import Peer
+from cockpit.protocol import CockpitProblem
+from cockpit.router import Router, RoutingRule
+from cockpit.transports import StdioTransport
+
+logger = logging.getLogger(__name__)
+
+
+def ensure_ferny_askpass() -> Path:
+    src_path = importlib.resources.files(ferny.__name__) / 'interaction_client.py'
+    src_data = src_path.read_text()
+
+    xdg_data_home = os.environ.get('XDG_DATA_HOME')
+    if xdg_data_home is None:
+        xdg_data_home = os.path.expanduser('~/.local/share')
+    dest_path = Path(xdg_data_home, 'cockpit-client-askpass')
+
+    logger.debug("Checking if %s exists...", dest_path)
+
+    # Check first to see if we already wrote the current version
+    try:
+        if dest_path.read_text() != src_data:
```

Comment:
Feels a bit strange to read and compare this as text instead of binary (isn't that an xz beipack? Or already uncompressed)? But even if it's text, comparing as binary avoids the UTF-8 decoding on both sides.

Reply:
Uh ya, you're absolutely right.  Not sure what I was thinking there.

Reply:
ah no.  Now I know what I was thinking.  This isn't the beipack — it's the ferny interaction client, which absolutely is text.

Reply:
I can change it to `bytes` anyway, though, to save ourselves the round-trip through unicode...